docs/faq.md

```diff
@@ -109,3 +110,9 @@ typings/{main,browser}.d.ts
 ```
 
 Where `typings/{main,browser}.d.ts` is a collection of references to installed definitions. Main and browser typings are written to separate directories for `tsconfig.json` exclude support - you can completely exclude both the main or browser typings.
+
+## Types of Typings
+
+There are two major types of type definitions - external modules and ambient definitions. Typings can install both. It supports modules by default, and also supports ambient declarations using the `--ambient` flag.
+
+An module definition is considered "external" if it has an `import` or `export` declaration at the top-level. Everything else can be considered "ambient". In the past, without tooling such as Typings, we've relied on writing ambient definitions to define modules using the `declare module '...'` syntax. With Typings, it ends up the same (wrapped in `declare module '...'`), but since the Typings tooling is doing the wrapping it can avoid namespace conflicts _and_ ensure nothing is leaking from ambient definitions into the global namespace.
```

Comment:
Besides namespace conflict, I learn that versioning is an even more important benefit. Should we mention it?

Also, was asking TS team on about the term for `declare module '...'`. Probably will not use it right now but in the future. 
Here is the reference;
https://github.com/Microsoft/TypeScript-Handbook/issues/242